lib/elixir/lib/enum.ex

```diff
@@ -803,11 +803,13 @@ defmodule Enum do
   @spec filter_map(t, (element -> as_boolean(term)),
     (element -> element)) :: list
 
-  def filter_map(enumerable, filter, mapper) when is_list(enumerable) do
+  def filter_map(enumerable, filter, mapper)
+  when is_list(enumerable) and is_function(filter, 1) and is_function(mapper, 1) do
```

Comment:
@whatyouhide I thought I had already convinced @lexmag the indented `with` is better. :P